pkg/network/dhcp/bridge.go

```diff
@@ -1,15 +1,45 @@
 package dhcp
 
 import (
+	"github.com/vishvananda/netlink"
+
+	v1 "kubevirt.io/client-go/api/v1"
 	"kubevirt.io/kubevirt/pkg/network/cache"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+	virtnetlink "kubevirt.io/kubevirt/pkg/network/link"
 )
 
 type BridgeConfigGenerator struct {
+	handler          netdriver.NetworkHandler
 	podInterfaceName string
 	cacheFactory     cache.InterfaceCacheFactory
 	launcherPID      string
+	vmiSpecIfaces    []v1.Interface
+	vmiSpecIface     *v1.Interface
 }
 
 func (d *BridgeConfigGenerator) Generate() (*cache.DHCPConfig, error) {
-	return d.cacheFactory.CacheDHCPConfigForPid(d.launcherPID).Read(d.podInterfaceName)
+	dhcpConfig, err := d.cacheFactory.CacheDHCPConfigForPid(d.launcherPID).Read(d.podInterfaceName)
```

Comment:
I like how this ended up, but I wonder if we shouldn't make explicit that this DHCP cache is actually a way to send the original MAC address from one process to another. 

```golang
mac, err := d.cacheFactory.CachePodLinkForPID(d.launcherPID).Read(d.podInterfaceName)
...
```

Follow-up would be OK for me.

Reply:
Right. I'm OK with tackling that in a follow up. Since we're doing this to increase the readability of the code, I think this would take it a tiny step further.